File: test/data/find_references/refs.py
x = 10

def process(x):
    y = x + 1
    return y

def transform(data):
    x = data * 2
    result = process(x)
    return result

class Service:
    def __init__(self, db):
        self.db = db

    def fetch(self, query):
        return self.db.execute(query)

    def run(self):
        data = self.fetch("SELECT 1")
        return process(data)

def main():
    svc = Service("postgres")
    result = svc.run()
    print(result)

File: test/data/find_references/refs.js
const config = { timeout: 30 };

function helper(x) {
    return x * 2;
}

function process(data) {
    const x = helper(data);
    console.log(x);
    return x;
}

class Service {
    constructor(db) {
        this.db = db;
    }

    fetch(query) {
        return this.db.execute(query);
    }

    run() {
        const data = this.fetch("SELECT 1");
        return process(data);
    }
}

const svc = new Service("postgres");
const result = svc.run();
console.log(result);
